backend/src/test/java/com/google/blackswan/mock/SimpleAlertGeneratorTest.java

```diff
@@ -0,0 +1,117 @@
+    "2019-07-21", 73, 
+    "2019-07-28", 59, 
+    "2019-08-04", 75,
+    "2019-08-11", 79,
+    "2019-08-18", 67
+  );
+  private static final int SET_THRESHOLD_LOW = 2;
+  private static final int SET_DATAPOINTS = 2;
+  private static final int EXPECTED_ALERT_SIZE = 2;
+  private static final AlertGenerator alertGenerator = new SimpleAlertGenerator(
+    SimpleAnomalyGenerator.createGeneratorWithString(
+      inputForAnomalyGenerator(), SET_THRESHOLD_LOW, SET_DATAPOINTS
+    )
+  );
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() throws Exception {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void getAlerts_returnsListOfAlerts() {
+    // Create first expected alert which contains one anomaly in July.
+    ImmutableMap<Timestamp, MetricValue> expectedDataPoints1 = ImmutableMap.of(
+      new Timestamp("2019-07-21"), new MetricValue(73),
+      new Timestamp("2019-07-28"), new MetricValue(59), 
+      new Timestamp("2019-08-04"), new MetricValue(75)
+    );
+    Anomaly expectedAnomalyGroup1 = new Anomaly(
+      new Timestamp("2019-07-21"), METRIC_NAME, DIMENSION_NAME, expectedDataPoints1
+    );
+    Alert expectedAlertJuly = new Alert(
+      new Timestamp("2019-08-01"), 
+      Arrays.asList(expectedAnomalyGroup1), 
+      Alert.StatusType.UNRESOLVED
+    );
+    // Create second expected alert which contains two anomalies that occurs in August. 
+    ImmutableMap<Timestamp, MetricValue> expectedDataPoints2 = ImmutableMap.of(
+      new Timestamp("2019-07-21"), new MetricValue(73),
+      new Timestamp("2019-07-28"), new MetricValue(59),
+      new Timestamp("2019-08-04"), new MetricValue(75), 
+      new Timestamp("2019-08-11"), new MetricValue(79), 
+      new Timestamp("2019-08-18"), new MetricValue(67)
+    );
+    Anomaly expectedAnomalyGroup2_1 = new Anomaly(
+      new Timestamp("2019-08-04"), METRIC_NAME, DIMENSION_NAME, expectedDataPoints2
+    );
+    ImmutableMap<Timestamp, MetricValue> expectedDataPoints3 = ImmutableMap.of(
+      new Timestamp("2019-07-28"), new MetricValue(59),
+      new Timestamp("2019-08-04"), new MetricValue(75), 
+      new Timestamp("2019-08-11"), new MetricValue(79), 
+      new Timestamp("2019-08-18"), new MetricValue(67)
+    );
+    Anomaly expectedAnomalyGroup2_2 = new Anomaly(
+      new Timestamp("2019-08-11"), METRIC_NAME, DIMENSION_NAME, expectedDataPoints3
+    );
+    Alert expectedAlertAugust = new Alert(
+      new Timestamp("2019-09-01"), 
+      Arrays.asList(expectedAnomalyGroup2_1, expectedAnomalyGroup2_2), 
+      Alert.StatusType.UNRESOLVED
+    );
+
+    List<Alert> generatedAlerts = alertGenerator.getAlerts();
+
+    assertEquals(generatedAlerts.size(), EXPECTED_ALERT_SIZE);
+    assertEquals(generatedAlerts.get(0), expectedAlertJuly);
+    assertEquals(generatedAlerts.get(1), expectedAlertAugust);
```

Comment:
assertEquals(Object expected, Object actual)
Asserts that two objects are equal.


and all the other non-object primitives, the assertEquals are in the format expected first and then actual later.

Reply:
thanks for letting me know! I will fix this from now on.